ocaml/testsuite/tests/typing-layouts-float64/basics_alpha.ml

```diff
@@ -697,3 +722,67 @@ Error: This expression has type t_float64
        but an expression was expected of type ('a : value)
        t_float64 has layout float64, which is not a sublayout of value.
 |}];;
+
+(***********************************************************)
+(* Test 14: unboxed float records work like normal records *)
+
+module FU = Stdlib__Float_u
+
+type t14_1 = { x : float#; y : float# }
+
+(* pattern matching *)
+let f14_1 {x;y} = FU.sub x y
+
+(* construction *)
+let r14 = { x = FU.of_float 3.14; y = FU.of_float 2.72 }
+
+let sum14_1 = FU.to_float (f14_1 r14)
+
+(* projection *)
+let f14_2 ({y;_} as r) = FU.sub r.x y
+
+let sum14_2 = FU.to_float (f14_1 r14)
+
+type t14_2 = { mutable a : float#; b : float#; mutable c : float# }
+
+let f14_3 ({b; c; _} as r) =
+  (* pure record update *)
+  let r' = { r with b = FU.of_float 20.0; c = r.a } in
+  (* mutation *)
+  r.a <- FU.sub r.a r'.b;
+  r'.a <- FU.of_float 42.0;
+  r'
+
+let a, b, c, a', b', c' =
+  let r = {a = FU.of_float 3.1; b = FU.of_float (-0.42); c = FU.of_float 27.7 } in
+  let r' = f14_3 r in
+  FU.to_float r.a,
+  FU.to_float r.b,
+  FU.to_float r.c,
+  FU.to_float r'.a,
+  FU.to_float r'.b,
+  FU.to_float r'.c
+
+let f14_4 r =
+  let {x; y} = r in
+  FU.add x y
+
+
+[%%expect{|
+module FU = Stdlib__Float_u
+type t14_1 = { x : float#; y : float#; }
+val f14_1 : t14_1 -> float# = <fun>
+val r14 : t14_1 = {x = <abstr>; y = <abstr>}
+val sum14_1 : float = 0.419999999999999929
+val f14_2 : t14_1 -> float# = <fun>
+val sum14_2 : float = 0.419999999999999929
+type t14_2 = { mutable a : float#; b : float#; mutable c : float#; }
+val f14_3 : t14_2 -> t14_2 = <fun>
+val a : float = -16.9
+val b : float = -0.42
+val c : float = 27.7
+val a' : float = 42.
+val b' : float = 20.
+val c' : float = 3.1
+val f14_4 : t14_1 -> float# = <fun>
+|}]
```

Comment:
Add a test for heterogeneous polymorphic equality?

Reply:
I have added a test (I put it over in `unboxed_floats.ml`, so that it runs in both bytecode and native code).  Though I'm not sure we actually want to make any promises about the behavior of this.